packages/transpiler/src/solc/schemas/ast.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * Schema of an Abstract Syntax Tree Node.
+ * This schema only describes the properies that are mandatory across all nodes.
+ * Additional attrs can exist depending on the entity the Node represents.
+ */
+
+'use strict';
+
+// A fully qualified, minimal object for this Schema is:
+/*
+{
+	"nodeType": "Literal",
+}
+*/
+
+const array = { type: 'array' },
+  number = { type: 'number' },
+  bool = { type: 'boolean' },
+  string = { type: 'string' },
+  object = { type: 'object' },
+  attrNull = { type: 'null' };
+
+export default {
```

Comment:
Yes, that is super loose schema. I've added it with a goal in mind to fail early in case if AST files are corrupt for whatever reason. Also, I think we should validate JSON as a general rule rather to trust it to to be what we expect.